src/functions.rs

```diff
@@ -207,14 +219,21 @@ pub fn is_executable(path: impl AsRef<Path>) -> io::Result<bool> {
 ///
 /// ```
 /// use permissions::access_syscall;
-/// use permissions::consts::{R_OK, W_OK, X_OK, F_OK};
+/// use permissions::consts::{R_OK, W_OK, F_OK};
+/// #[cfg(unix)]
+/// use permissions::consts::X_OK;
 ///
 /// fn main() -> std::io::Result<()> {
 ///     assert!(access_syscall("src/lib.rs", R_OK | W_OK)?);
+///     #[cfg(unix)]
 ///     assert!(access_syscall("/", R_OK | X_OK)?);
+///     #[cfg(windows)]
+///     assert!(access_syscall("/", R_OK)?);
 ///     assert!(access_syscall(".", F_OK)?);
 ///
+///     #[cfg(unix)]
 ///     assert!(!access_syscall("src/lib.rs", X_OK)?);
+///     #[cfg(unix)]
 ///     assert!(!access_syscall("/root", W_OK)?);
```

Comment:
This /root can be replaced by `/` to make this cross-platform, so you can remove the `cfg`.

Reply:
We actually cannot. The winapi returns true for this call, which is really odd.

I checked the implementation of another crate and it's awful:
https://github.com/Freaky/faccess/blob/master/src/lib.rs#L126-L375

But that's nothing new. So it's not a change from previous versions.